### DISCUSSION TITLE
Source code viewer's output is difficult to read in 1.47.5b7-b8 on dark theme

### DISCUSSION THREAD

**Discussion by u-RraaLL** [uBlockOrigin/uBlock-issues#2542] [General] [answered]
Moving discussion outside [commit comments](https://github.com/gorhill/uBlock/commit/e2dd008388340a3cc18d57c959ee9d3d6ace6d02#commitcomment-1041038180) per suggestion.

>>>>    The new colors on Dark Mode now make code difficult to read.
>>>
>>> You are talking about the code viewer only?
>>>
>> Yes.
>>
>>> I used an official CodeMirror theme as I do not want to go through all the coloring stuff again. You can suggest one from there: https://codemirror.net/5/demo/theme.html.
>>
>> I went though the themes, they're all readable. They're not what I see in b7 (and now b8).
>>
>> <details><summary>This is what I see in b7-8:</summary>
>>
>> ![image](https://user-images.githubusercontent.com/92610420/224551277-aea444ec-ff95-4bd1-9a64-50dba1ae1ac0.png)</details>
>> <details><summary>This is what I see in b6:</summary>
>>
>> ![image](https://user-images.githubusercontent.com/92610420/224551534-6806dae3-f829-434c-9e72-7297d5923d2a.png)</details>
>>
>> Though, since you're asking, my personal favorite would be `seti`.
>
> I'm going to look into this. We should move to a discussion though, here is not ideal.

**Comment by gorhill** [comment, marked as answer]
Should be fixed in 1.47.5b9

**Comment by Yuki2718** [comment]
Also highlights in asset-viewer is difficult to read:

<details>
<summary>Screenshot</summary>

![dark](https://user-images.githubusercontent.com/58900598/229366042-92af1c81-5ca9-44ef-a12c-4d81da1a74fa.png)

</details>

**Reply by peace2000** [reply]
I agree, what color would you consider as better option?

Imo blue looks good:

<details>
<summary>Screenshot</summary>

![kuva](https://user-images.githubusercontent.com/17256841/229367037-3eaa4a24-6dda-4f97-b1f6-3985533c69a1.png)

</details>

E.g. `#0034feb5`

**Reply by Yuki2718** [reply]
Yup, looks nice!

**Reply by gorhill** [reply]
~~CodeMirror uses a color with an alpha value, I discarded this alpha value when I worked on theme stuff. I just need to bring it back. The blue color could be mistaken as a selection, I find it confusing.~~

I just needed to ensure the `--cm-searching-ink` value used for text color of search matches was not being overridden.

**Reply by peace2000** [reply]
If blue isn't good, how about making the yellow color more fade?

<details>
<summary>Screenshot</summary>

![kuva](https://user-images.githubusercontent.com/17256841/229368950-5c1f794f-6cc0-4299-8eef-99b6b288b7a5.png)

</details>

`#fee3003d`

This option might not be so beautiful but I think that it's important to be able easily discern the highlighted text.

**Reply by gorhill** [reply]
There was already a color set for the text of search matches, it was being overridden while the original intention was to not have it overridden -- this is fixed in latest dev build.

**Reply by peace2000** [reply]
This looks a lot better!:

Latest dev:

<details>
<summary>Screenshot</summary>

![kuva](https://user-images.githubusercontent.com/17256841/229567964-69cd2e05-8483-4427-81b1-a69943802b28.png)

</details>

Latest stable:

<details>
<summary>Screenshot</summary>

![kuva](https://user-images.githubusercontent.com/17256841/229568172-028c5db2-4663-4685-a183-22930255db62.png)

</details>

**Comment by u-RraaLL** [comment]
@gorhill there seems to be some glitch in code viewer's searchbar. The match count doesn't show up for me if I don't open dev tools (I used inspect element to see if the counter was missing in the tool).

If I refresh with dev tools open and try a new search, the counter will show up quickly,. If I close dev tools and repeat, the counter doesn't appear. Even if I use the arrows to switch between matches.

Didn't seem big enough to make a new issue, so reporting here instead.

**Reply by gorhill** [reply]
I've experienced this sometimes but I don't know how to reproduce when I want to try to reproduce. I don't think this has anything to do with dev tools.

**Reply by peace2000** [reply]
I can reproduce it. It seems that when you search something for the first time the match count won't show up. If I remove my search term and then search something again, the match count appears.

**Reply by peace2000** [reply]
I'm running Firefox 112.0 and uBO 1.48.5b10. Tested now uBO 1.48.9b1 and it suffers from the same issue.

**Reply by peace2000** [reply]
Can't repro on Chrome 111.0.5563.147 & uBO 1.48.5b5

**Reply by gorhill** [reply]
I can't reproduce as you describe it on Firefox, and I have also seen the issue in Chromium.

**Reply by peace2000** [reply]
Hmm... I could reproduce that with clean browser profile i guess I'll try a virtual machine Firefox installation...

**Reply by gorhill** [reply]
I currently have been able now to reproduce apparently reliably (in Firefox):

- Force a reload of the code viewer tab (assuming you have opened to view some source)
- Search for a term which does not exist (I use `toto`)
- Remove the failed search term and search for something which exist: the count does not show up

**Reply by peace2000** [reply]
I loaded my virtual machine and refreshed my Firefox installation to get as clean experience as possible, installed latest uBO dev and I can still reproduce this.

Can you try to open this in code viewer: `https://www.telsu.fi/js/t.js?v=254`

And search for "var" and then remove it completely and then search again?

<details>

![kuva](https://user-images.githubusercontent.com/17256841/231587681-fcebbe16-4a82-461a-8c8c-375855767ecb.png)

</details>

->

<details>

![kuva](https://user-images.githubusercontent.com/17256841/231587800-4b675283-a942-46ed-a096-cd45c16c9092.png)

</details>

Can you @u-RraaLL repro this by any of the mentioned means?

**Reply by peace2000** [reply]
> I currently have been able now to reproduce apparently reliably (in Firefox):
> 
>     * Force a reload of the code viewer tab (assuming you have opened to view some source)
> 
>     * Search for a term which does not exist (I use `toto`)
> 
>     * Remove the failed search term and search for something which exist: the count does not show up

This way I can repro also. Though I don't need to force reload the tab.

**Reply by gorhill** [reply]
> The match count doesn't show up for me if I don't open dev tools

Hmm you are right, when I have the issue, opening the dev tools makes it go away immediately. I am starting to wonder if this is a Firefox issue with CSS `:empty`, which is used to either hide or show the search count.

**Reply by gorhill** [reply]
So far appear to have fixed it on my side: <https://github.com/gorhill/uBlock/commit/b6a5424c8d641e51565c43e3bf7c708664f90b23>.

**Reply by u-RraaLL** [reply]
Tested on Linux and Windows. It's fixed for me too.

**Reply by peace2000** [reply]
I confirm uBO 1.48.9b2 fixed this issue for me as well.
